src/kernel/symbol_visibility.h

```diff
@@ -0,0 +1,18 @@
+#ifndef BITCOIN_KERNEL_SYMBOL_VISIBILITY_H
+#define BITCOIN_KERNEL_SYMBOL_VISIBILITY_H
+
+#if defined(bitcoinkernel_EXPORTS)
+  #if defined(_WIN32)
+    #define BITCOINKERNEL_EXPORT __declspec(dllexport)
+  #else
+    #define BITCOINKERNEL_EXPORT __attribute__ ((visibility ("default")))
+  #endif
+#elif defined(_WIN32) && !defined(BITCOINKERNEL_STATIC)
```

Comment:
The meaning of `BITCOINKERNEL_STATIC` here is overloaded. Just to differentiate the use-cases, I'd prefer to have a 3rd define here:

`#elif defined(_WIN32) && !defined(BITCOIN_BUILD_INTERNAL) && !defined(BITCOINKERNEL_STATIC)`

`BITCOINKERNEL_STATIC`: I'm a user of the kernel building against a static kernel.
`BITCOIN_BUILD_INTERNAL`: I'm a component of Core who's not building against the kernel at all, I just happen to be seeing this header.

The former should stay as-is for bitcoin-chainstate, the latter is what we should be defining for the other internal libs.

Reply:
> `BITCOIN_BUILD_INTERNAL`: I'm a component of Core who's not building against the kernel at all, I just happen to be seeing this header.

`BITCOIN_INTERNAL_BUILD`  was in my draft of the updated branch this morning :)

Will rework the PR shortly.

Reply:
Thanks! Reworked.